src/acrquery/azext_acrquery/tests/latest/test_acr_query.py

```diff
@@ -1,54 +0,0 @@
-# --------------------------------------------------------------------------------------------
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License. See License.txt in the project root for license information.
-# --------------------------------------------------------------------------------------------
-
-from azure.cli.testsdk import ScenarioTest
-from azure.cli.command_modules.acr._docker_utils import (
-    EMPTY_GUID
-)
-
-
-class AcrQueryTests(ScenarioTest):
-
-    def test_acrquery(self):
-        self.kwargs.update({
-            'registry_name': 'metadataunittest',
-            'resource_group': 'cabarker',
-            'repository_name': 'nginx',
-            'query': '"Manifests | limit 1"',
-            'username': 'user',
-            'password': 'password',
-            'password_name': 'password'
-        })
-
-        # Query with bearer auth and filter by count
-        credentials = self.cmd(
-            'acr credential show -n {registry_name} -g {resource_group}').get_output_in_json()
-
-        self.kwargs['username'] = credentials['username']
-        self.kwargs['password'] = credentials['passwords'][0]['value']
-
-        self.cmd(
-            'acr query -n {registry_name} -q {query} --username {username} --password {password} ',
-            checks=[self.check('count', 1)])
-
-        # Query with basic auth and filter by repository
-        token = self.cmd('acr login -n {registry_name} --expose-token').get_output_in_json()
-        self.kwargs['username'] = EMPTY_GUID
-        self.kwargs['password'] = token["accessToken"]
```

Comment:
Actually, you can try using the customized `Preparer` to avoid recording the `accessToken` for `acr login`.

> If not, I can remove it temporarily and we can replace with mocks until the feature flag is removed.

If this is only a one-time temporary need, it is also acceptable

Reply:
Preparer generates a new resource within a specified or new resource group correct? 

In our case, the resource would need to exist already prior to running the tests (and we would need to login to it).

Reply:
The test scenario we have is that we have a pre-created resources because the feature is blocked by default we need to enable it in a specific resources. Should we avoid using pre-created resources for testing?

Reply:
@CarolineNB Yes, please use the dynamically created resources in the test instead of pre-created resources to avoid the problem of dependent resources being deleted will cause the test to be failed